### DISCUSSION TITLE
Optional hourly chime where the hour is the number of times to chime (with in reason) - CLOSED

### DISCUSSION THREAD

**Discussion by eharrow** [astlouys/Pico-Green-Clock#11] [Ideas]
I have an old clock that on the hour chimes 1 for 1 o'clock, 2 for 2 o'clock etc. up to 12 and then back to 1 again.  I quite like the idea of a chime indicating the current time for the pico-green.  This being a 24 hour clock 1 chime at 13:00 etc. would be appropriate.

Another spin on this would be an optional single chime on the half-hour which again is how my old clock on the mantel piece works.

--Ewan

**Comment by astlouys** [comment]
Hi Ewan,

That's a good idea. I'll give it a try in next version.

**Comment by UltrashRicco** [comment]
Hello!

This is an existing feature on many mechanical wall clocks, to replicate the church bell ringing the hours. Fun fact, in Spain, many radio stations also play a hourly chime, with the number of chimes representing the current hour.

I like the idea too fot the Pico-Green-Clock, but I am not sure I personnally would like it long-term: I would try it but I believe it would quickly become irritating (to me).

I believe the feature should be enabled/disabled following the same existing Chime time on/Chime time off settings (now that I think of it, I wonder if this exists on mechanical clocks!).

**Comment by eharrow** [comment]
Woo hoo just seen the code drop and will try to build and deploy with hourly and half hourly.

**Reply by eharrow** [reply]
The half hourly chime works.  Now I need to wait 30 mins.

**Reply by astlouys** [reply]
Yes, the "light" half-hour chime works fine and I think it`s an interesting feature.

The "CHIME_HOUR_COUNT", however, is interesting from 1 O'Clock up to 4 O'Clock (!!). After that, you want to revert to the "normal" hourly chime ;-)

There may be some specific environments where it could be used... I'm curious to see comments from users about this feature.

**Reply by eharrow** [reply]
Works pretty well.  I changed it to try to "slow" it down a bit i.e. beep every second or so e.g. BEEP, REST, BEEP, REST, BEEP, REST where the beep and rest are the same duration but that does not quite work as well as I hoped.  I changed  `sound_queue_active(1000, Dum1UInt8);` but that doe not affect the rest period.  Any ideas?

--Ewan

**Reply by astlouys** [reply]
Hi Ewan.

The parameters of sound_queue_active() are: 1) Duration (in milliseconds) and 2) RepeatCount, as you have already understood. However, when there is a repeat count, each "sound" is separated from the previous one by a 50 milliseconds pause, simply to discriminate the sound from its predecessor.

To achieve what you want to do, you'll have to make a "for" loop, and in each loop iteration, add a "sound_queue_active(Duration_ms, 1)", then another "sound_queue_active(Duration_ms, SILENT)".

If "SILENT" replaces the repeat count, it means that we want a pause of the specified duration in milliseconds (there is no need to have a repeat count for a pause, separated itself by... a pause ! That's why in this specific case, the "RepeatCount" parameter is tweeked to allow this flexibility).

**Reply by astlouys** [reply]
Ewan,

I closed the pull request for "Chime Hour Count".

I made your change in the source code so that users can use it, and I gave you credit in the README.MD. However, as mentioned previously, I didn't make it the default configuration (even if your new setup for clock ringing is less "aggressive" than the previous version).

Thank you for your collaboration !

André

**Comment by eharrow** [comment]
Thanks André. I hadn’t actually meant for you to get the PR - I was just trying to use the github flow with my own fork but it kept notifying you :(

I am also playing around with a github action/workflow to build using GitHub so I expect you will receive those as well though I don’t know why you should until I opebn a PR.

— Ewan

**Reply by astlouys** [reply]
Hi Ewan,

The way I understood it, it was like you wanted me to "pull" (retrieve) your source code and merge it to mine on GitHub. I'm not sure I would have been able to do it given the restricted privileges and Password / Credential manager / Personal access token problems that I have (!!), but the Chime Hourly Count was more the concern...

But that's fine, I'm not really familiar with all the bits and bites of GitHub myself, so no worry and we'll see if I receive other workflow / actions from your side. I'll let you know and I'm always curious to see what other friends may have in mind / may try with the code.

Let me know what you come up with !

André
